### ISSUE TITLE
Warning about inflating halo size does not take into account `Flat` dimensions

### ISSUE THREAD

**Issue by navidcy** [CliMA/Oceananigans.jl#1956]
https://github.com/CliMA/Oceananigans.jl/blob/2bb159c74774309ca1c80004e8cf5b5ae195d9f8/src/Grids/automatic_halo_sizing.jl#L40-L44

For example

```Julia
julia> using Oceananigans
[ Info: Oceananigans will use 12 threads

julia> grid = RegularRectilinearGrid(size = (10, 20), extent=(1, 2), topology=(Bounded, Bounded, Flat))
RegularRectilinearGrid{Float64, Bounded, Bounded, Flat}
                   domain: x ∈ [-1.6190752442450216e-17, 0.9999999999999999], y ∈ [0.0, 2.0], z ∈ [0.0, 0.0]
                 topology: (Bounded, Bounded, Flat)
  resolution (Nx, Ny, Nz): (10, 20, 1)
   halo size (Hx, Hy, Hz): (1, 1, 0)
grid spacing (Δx, Δy, Δz): (0.1, 0.1, 0.0)

julia> model = NonhydrostaticModel(grid=grid, advection = UpwindBiasedFifthOrder())
┌ Warning: Inflating model grid halo size to (3, 3, 0) and recreating grid. The model grid will be different from the input grid. To avoid this warning, pass halo=(3, 3, 0) when constructing the grid.
└ @ Oceananigans.Grids ~/.julia/packages/Oceananigans/9o1xJ/src/Grids/automatic_halo_sizing.jl:41
NonhydrostaticModel{CPU, Float64}(time = 0 seconds, iteration = 0)
├── grid: RegularRectilinearGrid{Float64, Bounded, Bounded, Flat}(Nx=10, Ny=20, Nz=1)
├── tracers: (:T, :S)
├── closure: Nothing
├── buoyancy: SeawaterBuoyancy{Float64, LinearEquationOfState{Float64}, Nothing, Nothing}
└── coriolis: Nothing
```
while, because `topology=(Bounded, Bounded, Flat)` we need to pass `halo=(3, 3)` to grid.

**Comment by francispoulin** [comment]
Very good point!

**Comment by glwagner** [comment]
I think this is raised by #1666 and an attempt to fix it is made in #1667 (but #1667 is not finished yet).

**Comment by navidcy** [comment]
Sorry then, closing in favor of #1666
